public/app/plugins/datasource/grafana-azure-monitor-datasource/components/MonitorConfig.tsx

```diff
@@ -38,7 +46,16 @@ export const MonitorConfig: FunctionComponent<Props> = (props: Props) => {
         onCredentialsChange={onCredentialsChange}
         getSubscriptions={getSubscriptions}
         disabled={props.options.readOnly}
-      />
+      >
+        <DefaultSubscription
+          subscriptions={subscriptions}
+          credentials={credentials}
```

Comment:
Could you please completely decouple the subscription from credentials?

Default subscription is just a filed `subscriptionId` in JsonData. We need to remove it from `AzureCredentialsBase` ([here](https://github.com/grafana/grafana/blob/e8c2091537a0641d71c3790ec88b5ae34da57bc6/public/app/plugins/datasource/grafana-azure-monitor-datasource/types/types.ts#L38)), to make the interface look same as it defined in ADX ([here](https://github.com/grafana/azure-data-explorer-datasource/blob/42176219a831d41fc1e1999617e4686d43ca8b84/src/components/ConfigEditor/AzureCredentials.ts#L19)).

Reply:
I agree this makes sense. It could be a breaking change so some migration code may be needed.

Reply:
Data schema should not change. The field `subscriptionId` stays unchanged. `AzureCredentialsBase` is in-memory only, so changing it should not require any migration.

Reply:
I've looked into this for the past few Grafana versions and this should not be a breaking change 😊